### PR TITLE
[GH-535] Add non-breaking space before 'show description button'

### PR DESCRIPTION
**Pull request by lunaticmonk**
#### Summary
Adds a space between the "show description" description and the icon on a board.

image

#### Ticket Link
https://github.com/mattermost/focalboard/issues/535

### CODE REVIEW

**Comment by lunaticmonk** [comment]
@chenilim I have signed the CLA but I don't know how to re-trigger the workflows to let the cla/mattermost workflow pass. Can you please help me out?

**Comment by lunaticmonk** [comment]
image

Here is the screenshot of the fix.

**Comment by lunaticmonk** [comment]
@chenilim Can you please review?

image

File: webapp/src/components/viewTitle.tsx [hunk 1]
```diff
@@ -64,6 +64,7 @@ const ViewTitle = React.memo((props: Props) => {
                         }}
                         icon={<ShowIcon/>}
                     >
+                        &nbsp;
```
Comment (jespino) [inline webapp/src/components/viewTitle.tsx:67]: I don't love the use of `&nbsp;` here, it remove certain control of the styling, do you mind to add a `.ViewTitle .add-buttons .Icon` css that add a `margin-right`, that way we can make more fine grain styling of this.

**Review by jespino — CHANGES_REQUESTED** [review]
I prefer to go with css intead of `&nbsp` for this visual bug.

File: webapp/src/components/viewTitle.tsx (hunk 1 above)
Comment (jespino) [reply webapp/src/components/viewTitle.tsx:67]: by the way, the right place would be the file `webapp/src/components/viewTitle.scss`

**Comment by harshilsharma63** [comment]
@lunaticmonk just checking in if you have any update on this?

**Comment by lunaticmonk** [comment]
@harshilsharma63 @jespino Pushed the css changes. Please review once.

File: webapp/src/components/viewTitle.scss [hunk 2]
```diff
@@ -38,4 +38,8 @@
     > .description > * {
         flex-grow: 1;
     }
+
+    .non-breaking-space {
```
Comment (jespino) [inline webapp/src/components/viewTitle.scss:42]: This is not what I was expecting, I was thinking more in something like changing the already existing `.add-buttons .Icon` class and add there the `margin-right: 0.5em`, that way you don't need the `span` element at all, and you also fix the same problem with the `Hide` and `Emoji` icons.

**Review by jespino — CHANGES_REQUESTED** [review]

**Comment by sbishel** [comment]
@lunaticmonk Just checking in if you are still working on this.

**Comment by lunaticmonk** [comment]
@sbishel Yes, will update this PR soon, was a bit busy from few days at work!

**Comment by lunaticmonk** [comment]
@sbishel I accidently logged out of my account on my local setup and can't remember the password. Is there a way to reset the password via some hack? I need to test these changes once. I also tried signing up again but getting:

image


Can you help me in this?

**Comment by sbishel** [comment]
@lunaticmonk Sorry, I missed this message. No, we don't have a hack for that. You could hack the change password code (ie, don't require old password) and run via ChangePassword API. 

I'm not sure why you can't register a new user are you seeing any other errors?

 If you want to push it up this PR, I'm happy to test.

**Comment by lunaticmonk** [comment]
@sbishel I tried doing few hacks but the page gets redirected to login and I cannot reset password. Looks like I will need your help in testing this and merging it. Meanwhile, I will set up the repo again in my system.

**Comment by sbishel** [comment]
@lunaticmonk Sure, merge your changes to this PR and I'll test. The only hack would be if your running source code to put a break point in where login fails and go around it.

**Comment by lunaticmonk** [comment]
@sbishel Didn't get you? Which PR should I merge my changes to? Is there any other branch you have where I should merge these?

**Comment by sbishel** [comment]
@lunaticmonk This one. I thought you had additional updates to this PR based on Jesus feedback.

**Comment by lunaticmonk** [comment]
@sbishel These are the only changes I believe. You can test this branch.

**Comment by sbishel** [comment]
Fixed by https://github.com/mattermost/focalboard/pull/765

**Pull request closed without merging**